### ISSUE TITLE
Refactor `sendValidationEmail` in `identity/api.ts` to Okta

### ISSUE THREAD

**Issue by JamieB-gu** [guardian/frontend#26349]
Refactor usage of `fetch` to use either Okta or cookies depending on the experiment variant. Example of this here:

https://github.com/guardian/dotcom-rendering/pull/8211/files?diff=unified&w=1#diff-ae460a81df60e9adb882cddbba675359ef3ef98f869a762f3598024fd1fa55a4L132-R160

New functions added in https://github.com/guardian/frontend/pull/26311 may be useful for this.

https://github.com/guardian/frontend/blob/22c84f55d331cfefd707b27f79adfeecad2a0828/static/src/javascripts/projects/common/modules/identity/api.ts#L365-L385

**Comment by coldlink** [comment]
This was an endpoint I was unaware of. Do we know how/where this is being used?

**Comment by coldlink** [comment]
As far as I can see this is the only usage of it here: https://github.com/guardian/frontend/blob/b7b68ce21a214a4b584d6c3d026b4d366fa6e000/static/src/javascripts/projects/common/modules/identity/validation-email.js#L23

which looks for elements with the `js-id-send-validation-email` class and adds an event listener on to it?

As for where that class appears I see it in 3 places.

1. [`accountDeletionBlock.scala.html`](https://github.com/guardian/frontend/blob/71168427fe3ceedec300a72f505365bc31515eb9/identity/app/views/profile/deletion/accountDeletionBlock.scala.html#L74)
2. [`accountDeletionForm.scala.html`](https://github.com/guardian/frontend/blob/912e3786b9ead20c8360dd60b51e42616423bc0e/identity/app/views/profile/deletion/accountDeletionForm.scala.html#L140)
3. [comment-box.js](https://github.com/guardian/frontend/blob/e4374df297a3a800ac34756a1ad290cdeca8cbc3/static/src/javascripts/projects/common/modules/discussion/comment-box.js#L104-L108
)


With the former 2, I'm currently in the process of migrating the account deletion flows out of frontend/identity to gateway, so I can handle those then.

With 3 I'll need to investigate further if we should still be using this email validation flow!

Essentially this is using an old way of verifying users, which is slightly insecure.

There are now only 2 ways to verify users securely:
- On account creation through https://profile.theguardian.com/register and setting a password using the link that was sent.
- On password reset when going through https://profile.theguardian.com/reset-password and resetting the user's password

So my thinking with 3 is that if a user does not have a validated email, we should send them to the reset password page instead.

**Comment by georgeblahblah** [comment]
I've opened a PR @ https://github.com/guardian/frontend/pull/26394 to note that the function is deprecated. Once the investigation is complete, we can open a new ticket to redirect to password reset from `comment-box.js`.

**Comment by ShraddhaPande** [comment]
@coldlink Is there anything you need to do to help unblock this ?

**Comment by coldlink** [comment]
This link/button should now redirect to the reset password page (https://profile.theguardian.com/reset-password) after investigation. This functionality has been deprecated.
